processing/src/main/java/io/druid/segment/data/LongSupplierSerializer.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.data;
+
+import com.google.common.io.ByteSink;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ */
+public interface LongSupplierSerializer extends Closeable
```

Comment:
This class should get some more javadocs that answer stuff like:

- what does `closeAndConsolidate` mean and how is that different from `close`?
- what does `writeToChannel` do, and should that be called before or after closing?
- when should `add` be called relative to other methods?
- when should `size` and `getSerializedSize` be called relative to other methods?
- what units does `size` return (bytes, number of longs)?